model/flow/version_beacon.go

```diff
@@ -155,3 +156,43 @@ func (v *VersionBeacon) String() string {
 	}
 	return buffer.String()
 }
+
+// ProtocolStateVersionUpgrade is a service event emitted by the FlowServiceAccount
+// to signal an upgrade to the Protocol State version. `NewProtocolStateVersion`
+// must be strictly greater than the currently active Protocol State Version,
+// otherwise the service event is ignored.
+// If the node software supports `NewProtocolStateVersion`, then it uses the
+// specified Protocol State Version, beginning with the first block B where BOTH:
+//  1. The `ProtocolStateVersionUpgrade` service event has been sealed in B's ancestry
+//  2. B.view >= `ActiveView`
```

Comment:
You are using the same symbol `B` for different meanings, which is usually confusing for readers. I would suggest to use a different symbol here:
```suggestion
// specified Protocol State Version, beginning with the first block `X` where BOTH:
//  1. The `ProtocolStateVersionUpgrade` service event has been sealed in B's ancestry
//  2. X.view >= `ActiveView`
```